### PR TITLE
Add REQUIRE_EMAIL_CONFIRM setting

### PR DESCRIPTION
**Pull request by windy1**
Signed-off-by: Walker Crouse <walkercrouse@hotmail.com>

### CODE REVIEW

File: spongeauth/accounts/middleware.py [hunk 1]
```diff
@@ -16,13 +17,13 @@ def __call__(self, request):
...
     def must_verify(user):
-        return user.is_authenticated() and not user.email_verified
+        return user.is_authenticated() and not user.email_verified and settings.REQUIRE_EMAIL_CONFIRM
 
     @staticmethod
     def may_pass(url):
         try:
             return getattr(
-                resolve(url).func, 'allow_without_verified_email', False)
+                resolve(url).func, 'allow_without_verified_email', False) or not settings.REQUIRE_EMAIL_CONFIRM
```
Comment (lukegb) [inline spongeauth/accounts/middleware.py]: I don't think you need this here as well.

File: spongeauth/spongeauth/settings/prod.py [hunk 2]
```diff
@@ -11,6 +11,8 @@
 
 SECRET_KEY = os.environ['SECRET_KEY']
 
+REQUIRE_EMAIL_CONFIRM = True
```
Comment (lukegb) [inline spongeauth/spongeauth/settings/prod.py]: If this is true in base it's probably not worthwhile setting it to `True` here.

**Review by lukegb — CHANGES_REQUESTED** [review]
Can you also add a test to

https://github.com/lukegb/SpongeAuth/blob/0295e712ee0a650b86242d9d9a681912c9c72580/spongeauth/accounts/tests/test_middleware_enforce_verified_emails.py

in `TestMustVerify` to make sure that it returns `False` if `REQUIRE_EMAIL_CONFIRM` is `False`, even if the other conditions are met?

**Comment by windy1** [comment]
@lukegb Updated

**Review by lukegb — APPROVED** [review]

**Comment by lukegb** [comment]
Thanks!

**Pull request merged**